app/controllers/favorites_controller.rb

```diff
@@ -2,4 +2,20 @@ class FavoritesController < ApplicationController
   def index
     @favorite_artists = current_user.user_artists
   end
+
+  def new
+    if UserArtist.create(user_id: current_user[:id], artist_id: params["id"])
+      flash[:success] = "#{find_artist_by_id(params["id"]).name} has been added to your favorites"
+      redirect_to request.referrer
```

Comment:
Never knew about this .referrer rails method!

Reply:
never seen it before either!